crates/hdx_parser/src/traits/declarations.rs

```diff
@@ -1,26 +1,31 @@
 use hdx_atom::{atom, Atom};
-use hdx_lexer::{Include, Kind, Token};
+use hdx_lexer::{Include, Kind};
 
-use crate::{discard, expect, expect_ignore_case, match_ignore_case, parser::Parser, unexpected, Result};
+use crate::{discard, expect, expect_ignore_case, parser::Parser, unexpected, Result};
 
 use super::Parse;
 
 pub trait Declaration<'a>: Sized + Parse<'a> {
 	type DeclarationValue: DeclarationValue<'a>;
 
 	fn parse_name(parser: &mut Parser<'a>) -> Result<Atom> {
-		match parser.next().clone() {
-			Token::Ident(atom) => {
+		let token = parser.next();
+		match token.kind() {
+			Kind::Ident => {
 				expect!(parser.next(), Kind::Colon);
-				Ok(atom.to_ascii_lowercase())
+				Ok(parser.parse_atom_lower(token))
 			}
-			token => unexpected!(parser, token),
+			_ => unexpected!(parser, token),
 		}
 	}
 
 	fn parse_important(parser: &mut Parser<'a>) -> Result<bool> {
-		if matches!(parser.peek(), Token::Delim('!'))
-			&& match_ignore_case!(parser.peek_n(2), Token::Ident(atom!("important")))
+		let peeked = parser.peek();
+		let nexted_peek = parser.peek_n(2);
+		if matches!(peeked.kind(), Kind::Delim)
+			&& matches!(peeked.char(), Some('!'))
+			&& matches!(nexted_peek.kind(), Kind::Ident)
+			&& matches!(parser.parse_atom_lower(*nexted_peek), atom!("important"))
```

Comment:
Is this ok? Should this have a macro or is there a better way?

Reply:
So you should be able to do something like:

```suggestion
		if match_delim!(parser.peek(), Kind::Delim('!'))
		  && match_ignore_case!(parser.peek_n(2), Kind::Ident(atom!("important"))
```

But I think for this to work we'd need to do some work to both the match macros used.